storage/src/adb/any/sync/resolver.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{
+    adb::{
+        any::{sync::Error, Any},
+        operation::Operation,
+    },
+    index::Translator,
+    mmr::verification::Proof,
+};
+use commonware_cryptography::Hasher;
+use commonware_runtime::{Clock, Metrics, Storage};
+use commonware_utils::Array;
+use futures::channel::oneshot;
+use std::{future::Future, num::NonZeroU64};
+
+/// Result of a call to [Resolver::get_proof].
+pub struct GetProofResult<H: Hasher, K: Array, V: Array> {
+    /// Proof that the operations are valid.
+    pub proof: Proof<H::Digest>,
+    /// The operations in the requested range.
+    pub operations: Vec<Operation<K, V>>,
+    /// A channel to send the result of the proof verification.
+    /// Caller should send `true` if the proof is valid, `false` otherwise.
+    /// Caller should ignore error if the channel is closed.
+    pub success_tx: oneshot::Sender<bool>,
+}
+
+/// Trait for network communication with the sync server
+pub trait Resolver<H: Hasher, K: Array, V: Array> {
+    /// Request proof and operations starting from the given index into an [Any] database's
+    /// operation log. Returns at most `max_ops` operations.
+    fn get_proof(
```

Comment:
For now, add a link to the issue to support a server with a growing tree (as `//`)